pkg/download/dependency_resolution/resolver/ahocorasick_dep_resolver.go

```diff
@@ -103,12 +105,8 @@ func findAndReplaceIDs(apiName string, configToBeUpdated config.Config, c depend
 			panic(fmt.Sprintf("No config found for given key %q", key))
 		}
 
-		if configToBeUpdated.Coordinate.Type == "dashboard" && conf.Coordinate.Type == "dashboard" {
-			continue // dashboards can not actually reference each other, but often contain a link to another inside a markdown tile
-		}
-
-		if conf.Coordinate == configToBeUpdated.Coordinate {
-			continue // skip self referencing configs
+		if !canReference(configToBeUpdated, conf) {
```

Comment:
if it can not reference, would a debug log message be a good idea here?

Reply:
While I agree that it could have some value, I don't necessarily think it would be a good idea. 
The basic resolver would print thousands on thousands of messages, as we're basically brute-forcing the dependencies. Each config will be compared with *all* configs, so 100 configs might produce 100 * 100 messages if they don't reference each other. 
For the faster algo, it will probably also provide a lot of spam, even though it might be improved.